### PR TITLE
[data-get][s]: data get r links from DataHub -refs #323

### PR DESCRIPTION
**Pull request by Mikanebu**
- data get r links from DataHub

### CODE REVIEW

File: bin/data-get.js [hunk 1]
```diff
@@ -69,27 +69,30 @@ const run = async () => {
...
+            #issue: https://github.com/datahq/datahub-qa/issues/86  */
+        const zipped_dataset_url  = `https://datahub.io/${owner}/${name}/r/${name}_zip.zip?jwt=${token}`
+        const archive_path = await saveFileFromUrl(zipped_dataset_url, 'zip')
+        // unzip archive into destination folder
+        fs.createReadStream(archive_path)
+          .pipe(unzip.Extract({ path: pathToSave }))
+          // removing the archive file once we extracted all the dataset files
+          .on('finish', () => {fs.unlinkSync(archive_path)})
+      } else {
+        pathToSave = await saveFileFromUrl(identifier, argv.format)
```
Comment (zelima) [inline bin/data-get.js]: It's really hard to follow the logic here. Does this mean it will download everything that has the identifier length not equal to 5? if so why it's not done after line 96 where it says `// If it is not a dataset - download the file`

To be honest `if (idParts.length === 5) {...}` smells very badly and think will lead to bunch of other bugs

File: bin/data-get.js [hunk 2]
```diff
@@ -134,6 +137,7 @@ const saveFileFromUrl = (url, format) => {
         err.message += ' or Forbidden.'
       }
       handleError(err)
+      process.exit(1)
```
Comment (anuveyatsu) [inline bin/data-get.js]: `process.exit(1)` should not be here as it's in `handleError` function now.

File: bin/data-get.js (hunk 1 above)
Comment (anuveyatsu) [reply bin/data-get.js]: I think logic should be:

```javascript
if (identifier.includes('/r/') {
  pathToSave = await saveFileFromUrl(identifier, argv.format)
} else {
  // previous logic
}
```

so we assume that if `/r/` is in identifier then it's `r` link.

**Review by anuveyatsu — CHANGES_REQUESTED** [review]
See comments above

File: lib/utils/error.js [hunk 3]
```diff
@@ -42,6 +42,7 @@ function handleError(err, {debug = false} = {}) {
       err.forEach(err => error(err.message))
     } else {
       error(err)
+      process.exit(1)
```
Comment (anuveyatsu) [inline lib/utils/error.js]: @Mikanebu can you please remove it? The `process.exit(1)` happens after a report is sent to Sentry. You're probably experiencing a buggy behaviour because you've setup `datahub=dev` which doesn't send reports so never hits `process.exit(1)` here - https://github.com/datahq/data-cli/blob/master/lib/utils/error.js#L25

Comment (anuveyatsu) [reply lib/utils/error.js]: or does it break tests?

Comment (Mikanebu) [reply lib/utils/error.js]: @anuveyatsu Spinner keeps loading 
```
> Error! 404: Not Found. Requested URL: https://datahub.io/core/co2-ppm/r/15.csv
test1
> Error! Cannot read property 'pipe' of undefined
⠋ Loading...
```

**Review by anuveyatsu — APPROVED** [review]
👍
